### ISSUE TITLE
wrong "gca" in plot

### ISSUE THREAD

**Issue by mathause** [regionmask/regionmask#316]
The logic to get the current axes in `Regions.plot` is wrong. It is currently implemented as:

https://github.com/regionmask/regionmask/blob/f14aa559036b84eb1b249390b83067506149f479/regionmask/core/plot.py#L238-L239

This leads to the following wrong behaviour:

```python
regionmask.defined_regions.natural_earth_v5_0_0.land_110.plot()
regionmask.defined_regions.prudence.plot()

plt.savefig("wrong_gca.png", dpi=300, bbox_inches="tight")
```

![wrong_gca](https://user-images.githubusercontent.com/10194086/151344290-e41fdb2b-e8c4-43d6-8dad-79f5427e9889.png)

**Comment by mathause** [comment]
This can be fixed as per the following snippet (which I once added in [xarray/plot/utils.py#L445](https://github.com/pydata/xarray/blob/10bfa77425c459691abac26477e25c5681dc396f/xarray/plot/utils.py#L445))




```python
def _maybe_gca(**kwargs):

    import matplotlib.pyplot as plt

    # can call gcf unconditionally: either it exists or would be created by plt.axes
    f = plt.gcf()

    # only call gca if an active axes exists
    if f.axes:
        # can not pass kwargs to active axes
        return plt.gca()

    return plt.axes(**kwargs)


f, ax = plt.subplots(1, 1)  # , subplot_kw=dict(projection=ccrs.Robinson()))

ax_new = _maybe_gca(projection=ccrs.PlateCarree())


assert ax is ax_new

from cartopy.mpl import geoaxes

if not isinstance(ax_new, geoaxes.GeoAxes):

    raise TypeError(
        "The current axes is not a cartopy GeoAxes. "
        "Either provide a GeoAxes or use ``plot_region``"
    )
```

**Comment by mathause** [comment]
Closed by #321
